src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -264,8 +272,27 @@ float64_t CCrossValidation::evaluate_one_run()
 		m_machine->set_store_model_features(true);
 
 		/* do actual cross-validation */
+#pragma omp parallel for
 		for (index_t i=0; i <num_subsets; ++i)
 		{
+			CMachine* machine;
```

Comment:
We copy the feature matrix this way, but that is OK for problems that are not too big. Let's see whether this works.
We can then start optimising further